activerecord/lib/active_record/connection_handling.rb

```diff
@@ -151,10 +151,10 @@ def connected_to(database: nil, role: nil, shard: nil, prevent_writes: false, &b
           role = role.to_sym
         end
 
-        db_config = resolve_config_for_connection(database)
+        db_config, owner_name = resolve_config_for_connection(database)
         handler = lookup_connection_handler(role)
 
-        handler.establish_connection(db_config)
+        handler.establish_connection(db_config, :default, owner_name)
```

Comment:
Should `:default` be `ActiveRecord::Base.default_pool_key`?

Reply:
Good catch, I fixed this and the other one.